Automattic/wp-calypso
# SiteAddressChanger:  Update variable & module naming

### Summary

This PR seeks to rename the remaining variable names, function names, file and directory names that are still making references to `SiteRenamer` or similar naming to the finalised `SiteAddressChanger` & `SiteAddressChange` language for the sake of consistency and clarity.

There should be no other changes that would have an effect on behaviour.

### Testing

Though there should be no changes to behaviour, it is very much worth giving the feature a full run through to ensure that the updates to naming throughout have not caused regressions along the way.

- Starting at http://calypso.localhost:3000/domains/manage/
- Select a site with a site address that you don't mind changing (and thus losing)
- Click through to see the site address changer UI @ `http://calypso.localhost:3000/domains/manage/{ that-site.wordpress.com }/edit/{ that-site.wordpress.com }`
-  Input a new name for your site (recommended something silly that future folk wouldn't want)
- Submit the form and continue through the confirmation dialog.
- Once you get confirmation that the site address has been changed, look through your list of sites.
  - You should not see the old site name in this list (because the discard property is now always `true`

fixes #24267
